tests/unit/mathematics/linalg/operations/Eigen3_operations_unittest.cc

```diff
@@ -465,6 +465,116 @@ TEST(LinalgBackendEigen, SGMatrix_cross_entropy)
 	EXPECT_NEAR(ref, result, 1e-15);
 }
 
+TEST(LinalgBackendEigen, SGMatrix_pinv_psd)
+{
+	float64_t A_data[] = {2.0, -1.0, 0.0, -1.0, 2.0, -1.0, 0.0, -1.0, 2.0};
+	// inverse generated by scipy pinv
+	float64_t scipy_result_data[] = {0.75, 0.5,  0.25, 0.5, 1.0,
+	                                 0.5,  0.25, 0.5,  0.75};
+
+	SGMatrix<float64_t> A(A_data, 3, 3, false);
+	SGMatrix<float64_t> result(scipy_result_data, 3, 3, false);
+
+	SGMatrix<float64_t> identity_matrix(3, 3);
+	linalg::identity(identity_matrix);
+	// using symmetric eigen solver
+	SGMatrix<float64_t> A_pinvh(3, 3);
+	linalg::pinvh(A, A_pinvh);
+	// using singular value decomposition
+	SGMatrix<float64_t> A_pinv(3, 3);
+	linalg::pinv(A, A_pinv);
+	SGMatrix<float64_t> I_check = linalg::matrix_prod(A, A_pinvh);
+	for (auto i : range(3))
+	{
+		for (auto j : range(3))
+		{
+			EXPECT_NEAR(identity_matrix(i, j), I_check(i, j), 1e-14);
+			EXPECT_NEAR(result(i, j), A_pinvh(i, j), 1e-14);
+			EXPECT_NEAR(result(i, j), A_pinv(i, j), 1e-14);
+		}
+	}
+	// no memory errors 
+	EXPECT_NO_THROW(linalg::pinvh(A, A));
+}
+
+TEST(LinalgBackendEigen, SGMatrix_pinv_2x4)
+{
+	SGMatrix<float64_t> A(2, 4);
```

Comment:
btw you should add a test for each method that complains if wrong dimensional result matrices are passed.
So that we have coverage of the REQUIRE statements